### PR TITLE
added abstraction layer `bootloader_env`

### PR DESCRIPTION
**Pull request by mlilien**
depends on https://github.com/omnect/concourse-pipelines/pull/118

### CODE REVIEW

File: src/bootloader_env/grub_env.rs [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+        .arg("list")
+        .output()
+        .with_context(|| "failed to execute 'grub-editenv list'")?;
+    ensure!(
+        list.status.success(),
+        "grub-editenv list: command returned with error"
+    );
+    let list = String::from_utf8(list.stdout)?;
+    let list = list.split('\n');
+    let mut value: String = "".to_string();
```
Comment (JanZachmann) [inline src/bootloader_env/grub_env.rs:18]: is `String`really needed?

File: src/bootloader_env/uboot_env.rs [hunk 2]
```diff
@@ -0,0 +1,46 @@
+use anyhow::{bail, ensure, Context, Result};
+use std::process::Command;
+
+pub fn bootloader_env(key: &str) -> Result<String> {
+    let value = Command::new("sudo").arg("fw_printenv").arg(key).output()?;
+    if !value.status.success() {
+        bail!("fw_printenv {} failed", key);
```
Comment (JanZachmann) [inline src/bootloader_env/uboot_env.rs:7]: just an info. I try to  format text always inline (if possible):
`bail!("fw_printenv {key} failed");`

File: src/bootloader_env/grub_env.rs (hunk 1 above)
Comment (mlilien) [reply src/bootloader_env/grub_env.rs]: done

File: src/bootloader_env/uboot_env.rs (hunk 2 above)
Comment (mlilien) [reply src/bootloader_env/uboot_env.rs]: done

**Review by JanZachmann — APPROVED** [review]

**Pull request merged**